include/sdf/Error.hh

```diff
@@ -179,6 +179,12 @@ namespace sdf
 
     /// \brief Generic error during parsing.
     PARSING_ERROR,
+
+    /// \brief Error trying to find a file.
+    FILE_NOT_FOUND,
+
+    /// \brief Error trying to access a pointer.
+    POINTER_ERROR,
```

Comment:
I agree on keeping changes to the `enum` to a minimal. 

I believe `FATAL_ERROR` will throw an exception when calling `throwOrPrintErrors` which will incur in a change of behavior, any other suggestion instead of `POINTER_ERROR`?

Reply:
I see only one place in `parser.cc` where `POINTER_ERROR` is proposed to be used (`checkCanonicalLinkNames`), and that function is only called in one place in our code (`src/gz.cc`), and it is passing an address to a local `sdf::Root` object, so I don't think the error is even possible in that case. Even so, if we call the new form of `checkCanonicalLinkNames` in that method and pass an `Errors` object to it, then `throwOrPrintErrors` wouldn't be called and it would avoid any behavior change associated with a `FATAL_ERROR`

Reply:
Sounds good, using `FATAL_ERROR` then.